### PR TITLE
Add categry exclusion filter

### PR DESCRIPTION
**Pull request by DrewScoggins**

### CODE REVIEW

File: src/harness/BenchmarkDotNet.Extensions/PerfLabExporter.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using BenchmarkDotNet.Disassemblers;
```
Comment (billwert) [inline src/harness/BenchmarkDotNet.Extensions/PerfLabExporter.cs:5]: looks like this snuck in from your other change?

File: src/harness/BenchmarkDotNet.Extensions/ExclusionFilter.cs [hunk 2]
```diff
@@ -27,4 +29,41 @@ public bool Predicate(BenchmarkCase benchmarkCase)
...
+
+        public bool Predicate(BenchmarkCase benchmarkCase)
+        {
+            if(patterns == null)
+            {
+                return true;
+            }
+            foreach (var category in benchmarkCase.Descriptor.Categories)
+            {
+                if(patterns.Any(pattern => category.ToLowerInvariant().Equals(pattern.userValue.ToLowerInvariant()) || pattern.regex.IsMatch(category)))
```
Comment (billwert) [inline src/harness/BenchmarkDotNet.Extensions/ExclusionFilter.cs:57]: why `ToLowerInvariant` instead of `Equals(str, StringComparison.OrdinalIgnoreCase)`?

Comment (DrewScoggins) [reply src/harness/BenchmarkDotNet.Extensions/ExclusionFilter.cs:57]: No reason other than I thought of the other way first. I like your way better.

File: src/harness/BenchmarkDotNet.Extensions/ExclusionFilter.cs [hunk 3]
```diff
@@ -27,4 +29,41 @@ public bool Predicate(BenchmarkCase benchmarkCase)
...
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        // https://stackoverflow.com/a/6907849/5852046 not perfect but should work for all we need
+        private static string WildcardToRegex(string pattern) => $"^{Regex.Escape(pattern).Replace(@"\*", ".*").Replace(@"\?", ".")}$";
```
Comment (adamsitnik) [inline src/harness/BenchmarkDotNet.Extensions/ExclusionFilter.cs:67]: similarly to `ExclusionFilter` from #1307 you should be able to reuse the existing  [AnyCategoriesFilter](https://github.com/dotnet/BenchmarkDotNet/blob/master/src/BenchmarkDotNet/Filters/AnyCategoriesFilter.cs) from BenchmarkDotNet and just negate the output:

```suggestion
        private readonly AnyCategoriesFilter categoriesFilter;

        public CategoryExclusionFilter(List<string> patterns) => categoriesFilter = new AnyCategoriesFilter(patterns.ToArray());

        public bool Predicate(BenchmarkCase benchmarkCase) => !categoriesFilter.Predicate(benchmarkCase);
```

Comment (DrewScoggins) [reply src/harness/BenchmarkDotNet.Extensions/ExclusionFilter.cs:67]: Sounds good.

**Comment by DrewScoggins** [comment]
FYI @SamMonoRT

**Pull request merged**